src/System.CommandLine.Tests/CompletionContextTests.cs

```diff
@@ -33,6 +36,128 @@ public void CommandLineText_preserves_command_line_prior_to_splitting_when_compl
                        .Be(commandLine);
         }
 
+        [Fact]
+        public void CommandLineText_is_parsed_when_option_other_than_last_is_in_name_equals_sign_value_format()
+        {
+           
+            CliRootCommand command = new CliRootCommand
+            {
+                new CliCommand("inner")
+                {
+                    new CliOption<string>("--optionOne"),
+                    new CliOption<string>("--optionTwo")
+                }
+            };
+
+            var commandLine = "inner --optionOne=argument1 --optionTwo argument2";
+
+            var parseResult = command.Parse(commandLine);
+            parseResult.GetCompletions();
+
+            Assert.True(parseResult.Tokens[1].Value == "--optionOne");
+            Assert.True(parseResult.Tokens[2].Value == "argument1");
+            Assert.True(parseResult.Tokens[3].Value == "--optionTwo");
+            Assert.True(parseResult.Tokens[4].Value == "argument2");
```

Comment:
We generally avoid the use of `Assert.IsTrue` in this codebase as it provides very little useful feedback when a test fails. Also, when there are multiple assertions, only the first failure is seen in the output, without feedback about whether the other assertions pass or fail.

I'd suggest changing the assertions to the following, which addresses both concerns:

```csharp
parseResult.Tokens.Should().BeEquivalentSequenceTo("--optionOne", "argument1", "--optionTwo", "argument2");
```

Reply:
I tried

`parseResult.Tokens.Should().BeEquivalentSequenceTo("--optionOne", "argument1", "--optionTwo", "argument2");`

and i get this error

Expected item[0] to be System.String, but found System.CommandLine.Parsing.CliToken

I also tried this

`            parseResult.Tokens.Should().BeEquivalentSequenceTo(
                new CommandLine.Parsing.CliToken("inner", CommandLine.Parsing.CliTokenType.Command, null),
                new CommandLine.Parsing.CliToken("--optionOne", CommandLine.Parsing.CliTokenType.Command, null),
                new CommandLine.Parsing.CliToken("argument1", CommandLine.Parsing.CliTokenType.Command, null),
                new CommandLine.Parsing.CliToken("--optionTwo", CommandLine.Parsing.CliTokenType.Command, null),
                new CommandLine.Parsing.CliToken("argument2", CommandLine.Parsing.CliTokenType.Command, null)
            );`
            
and got the following error

Expected object to be inner, but found inner.
Expected object to be --optionOne, but found --optionOne.
Expected object to be argument1, but found argument1.
Expected object to be --optionTwo, but found --optionTwo.
Expected object to be argument2, but found argument2.

Any ideas how to make this work?